src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.AspNetCore.Analyzers.WebApplicationBuilder.Fixers;
+public class WebApplicationBuilderFixer : CodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(new[]
+    {
+        // Add other diagnostic descriptor id's
+        DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id,
+        DiagnosticDescriptors.DoNotUseHostConfigureServices.Id,
+        DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id
+     });
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            string id = diagnostic.Id;
+            switch (id)
+            {
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Logging properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Logging"),
+                    equivalenceKey: DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id),
+                    diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Services properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Services"),
+                    equivalenceKey: DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                    diagnostic);
```

Comment:
```suggestion
                    context.RegisterCodeFix(
                        CodeAction.Create(
                            "Fix references to Services properties on WebApplicationBuilder",
                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Services"),
                            equivalenceKey: DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
                        diagnostic);
```
nit: Indent this and other registered code fixes so it's easier to tell what is passed to each method.